include/device.h

```diff
@@ -234,7 +239,7 @@ struct device_config {
  * @param driver_data driver instance data. For driver use only
  */
 struct device {
-	struct device_config *config;
+	const char *parent_name;
```

Comment:
I think we should also consider scenarios where there could be multiple device dependencies for the child device.

Reply:
Yes, I have in mind a I2S/I2C codec example. The codec could be controlled by I2C but needed by I2S. The I2C <-> Codec dependency will be traced by hierarchy, the I2S <-> Codec must be handled differently.